docs/internals/optimizer.rst

```diff
@@ -385,6 +385,20 @@ The preprocessing components perform transformations to get the program
 into a certain normal form that is easier to work with. This normal
 form is kept during the rest of the optimization process.
 
+Some of the steps in this group should be considered prerequisites for all other components.
+Currently these are:
+
+- :ref:`disambiguator`
+- :ref:`function-hoister`
+- :ref:`function-grouper`
+- :ref:`for-loop-init-rewriter`
```

Comment:
Can we reorder the detailed description of the steps so that it reflects this order? Right now it is `Disambiguator`, `FunctionHoister`, `FunctionGrouper`, `ForLoopConditionIntoBody`, and only then `ForLoopInitRewriter`